students/ZackConnaughton/lessons/lesson08/assignment/inventory.py

```diff
@@ -0,0 +1,28 @@
+"""
+Assignment 8 Norton Furniture Assignment
+Inventory Management System
+"""
+import csv
+from functools import partial
+
+def add_furniture(output_file, *args):
+    """
+    creates or appends to an invoice file the arguments passed
+    """
+    with open(output_file, "a+") as file:
+        str_args = [str(item) for item in args]
+        file.write(",".join(str_args))
+        file.write("\n")
+
+def rental_items(rental_file, customer="", output_file=""):
+    with open(rental_file) as file:
+        reader = csv.reader(file)
+        with open(output_file, "a+") as output_file:
+            for row in reader:
+                if row[0] == customer:
+                    output_file.write(",".join(row))
+                    output_file.write("\n")
+
+
+def single_customer(customer, output_file):
+    return partial(rental_items, customer=customer, output_file=output_file)
```

Comment:
This is close, but not quite what was asked for. There is no closure here, which was part of the directions. Notice that you are writing some of the same code in rental_items as you did in add_furniture. The idea was to change the signature of add_furniture, using partial, and then use that in the function that you return. So, there will be two functions total, not three. It is not efficient,  so you wouldn't do it in real life, but it teaches you to use partial and closures. :)